File: 전예진/BruteForce/prob5.py
from itertools import permutations

def solution(k, dungeons):
    all_dungeons = list(permutations(dungeons, len(dungeons)))

    max_cnt = 0

    for dungeon_order in all_dungeons:
        current_stamina = k
        current_dungeon_cnt = 0

        for dungeon in dungeon_order:
            min_required_stamina = dungeon[0]
            consumption_stamina = dungeon[1]

            if current_stamina >= min_required_stamina:
                current_stamina -= consumption_stamina
                current_dungeon_cnt += 1
            else:
                break

        max_cnt = max(max_cnt, current_dungeon_cnt)

    return max_cnt

File: 주다애/BruteForce/prob5.java
class Solution {
    boolean[] visited;
    int answer = -1;
    public int solution(int k, int[][] dungeons) {
        visited = new boolean[dungeons.length];
        recursion(0, k, dungeons);
        return answer;
    }
    
    public void recursion(int c, int k, int[][] dungeons) {
        int len = dungeons.length;
        for(int i = 0; i < len; i++) {
            if(visited[i] || dungeons[i][0] > k) {
                continue;
            }
            visited[i] = true;
            recursion(c + 1, k - dungeons[i][1], dungeons);
            visited[i] = false;
        }
        answer = Math.max(c, answer);
    }
}
